File: src/main/java/no/nav/domain/pensjon/regler/grunnlag/BarnDetalj.java
package no.nav.domain.pensjon.regler.grunnlag;

import no.nav.domain.pensjon.regler.PenPerson;

import java.io.Serializable;
import java.util.Date;

public class BarnDetalj implements Serializable {

    
    /**
     * Angir barnets andre forelder enn søker.
     */
    private PenPerson annenForelder;
    /**
     * Angir om barnet bor med en begge foreldre.
     * "true" betyr fellesbarn, "false" betyr særkullsbarn.
     */
    private boolean borMedBeggeForeldre;
    /**
     * Fra-og-med dato for når barnet bor sammen med begge foreldrene.
     */
    private Date borFomDato;
    /**
     * Til-og-med dato for når barnet bor sammen med begge foreldrene.
     */
    private Date borTomDato;

    /**
     * Angir om barnet har hatt inntekt over 1G.
     */
    private boolean inntektOver1G;
    /**
     * Angir hvorvidt barnet er under utdanning.
     */
    private boolean underUtdanning;

}

File: src/main/kotlin/no/nav/pensjon/regler/domain/grunnlag/BarnDetalj.kt
package no.nav.pensjon.regler.domain.grunnlag

import no.nav.pensjon.regler.domain.PenPerson
import java.io.Serializable
import java.util.*
import java.time.LocalDate

class BarnDetalj : Serializable {
    /**
     * Angir barnets andre forelder enn søker.
     */
    var annenForelder: PenPerson? = null

    /**
     * Angir om barnet bor med en begge foreldre.
     * "true" betyr fellesbarn, "false" betyr særkullsbarn.
     */
    var borMedBeggeForeldre = false

    /**
     * Fra-og-med dato for når barnet bor sammen med begge foreldrene.
     */
    @Deprecated("Use borFomDatoLd instead")
    var borFomDato: Date? = null
    var borFomDatoLd: LocalDate? = null

    /**
     * Til-og-med dato for når barnet bor sammen med begge foreldrene.
     */
    @Deprecated("Use borTomDatoLd instead")
    var borTomDato: Date? = null
    var borTomDatoLd: LocalDate? = null

    /**
     * Angir om barnet har hatt inntekt over 1G.
     */
    var inntektOver1G = false

    /**
     * Angir hvorvidt barnet er under utdanning.
     */
    var underUtdanning = false
}